server/services/teamsSummaryCountService.ts

```diff
@@ -228,39 +225,37 @@ export default class TeamsSummaryCountService {
    */
   async getTeamVeracodeVulnerabilityCounts(
     products: StrapiProduct[],
-    serviceCatalogueService: ServiceCatalogueService,
   ): Promise<{ veryHigh: number; high: number; medium: number; low: number }> {
     if (!Array.isArray(products) || products.length === 0) {
       return { veryHigh: 0, high: 0, medium: 0, low: 0 }
     }
     try {
-      const allComponents = await serviceCatalogueService.getComponents()
+      const allComponents = await this.serviceCatalogueService.getComponents()
       const productIds = new Set(products.map(p => p.id))
       const validComponents = allComponents.filter(component => {
-        const productId = component?.attributes?.product?.data?.id
+        const productId = component?.product?.id
         return productId && productIds.has(productId)
       })
 
-      const counts = validComponents.reduce(
-        (total, component) => {
-          const summary = component.attributes?.veracode_results_summary as VeracodeResultsSummary | undefined
-          if (!summary?.severity) return total
-
-          return summary.severity.reduce((outerTotal, severity) => {
-            return severity.category.reduce((innerTotal, category) => {
-              if (VALID_SEVERITIES.includes(category.severity as Severity)) {
-                return {
-                  ...innerTotal,
-                  [category.severity]: innerTotal[category.severity as Severity] + category.count,
-                }
-              }
-              logger.warn(`[getTeamVeracodeVulnerabilityCounts] Unexpected severity: ${category.severity}`)
-              return innerTotal
-            }, outerTotal)
-          }, total)
-        },
-        VALID_SEVERITIES.reduce((acc, sev) => ({ ...acc, [sev]: 0 }), {} as Record<Severity, number>),
-      )
+      // Initialize counts with all valid severities set to 0
+      const counts = Object.fromEntries(VALID_SEVERITIES.map(severity => [severity, 0]))
+
+      for (const component of validComponents) {
+        const summary = component.veracode_results_summary as unknown as VeracodeResultsSummary
+
+        const severities: [string, number][] = (summary?.severity || [])
+          .flatMap(severity => severity.category)
+          .map(category => [category.severity, category.count])
+
+        for (const [severity, count] of severities) {
+          if (VALID_SEVERITIES.includes(severity)) {
+            counts[severity] += count
+          } else {
+            logger.warn(`[getTeamVeracodeVulnerabilityCounts] Unexpected severity: ${severity}`)
+          }
+        }
+      }
+
```

Comment:
Hey @cugorji-justice , we ended up rewriting this a little bit as we had a bug in this area that was difficult to debug. It's equivalent functionality.
 
If you want to have a chat about it then let me know